core/src/main/scala/scalaz/EitherT.scala

```diff
@@ -330,6 +330,12 @@ sealed abstract class EitherTInstances2 extends EitherTInstances3 {
     new EitherTMonadError[F, E] {
       implicit def F = F0
     }
+
+  implicit def eitherTMonadReader[E, F[_], R](implicit F0: MonadReader[F, R], M0: Monad[F]): MonadReader[EitherT[E, F, ?], R] = 
```

Comment:
```
[error]  both method eitherTMonadReader in class EitherTInstances2 of type [E, F[_], R](implicit F0: scalaz.MonadReader[F,R], implicit M0: scalaz.Monad[F])scalaz.MonadReader[[γ$6$]scalaz.EitherT[E,F,γ$6$],R]
[error]  and method eitherTMonadError in class EitherTInstances2 of type [F[_], E](implicit F0: scalaz.Monad[F])scalaz.MonadError[[γ$5$]scalaz.EitherT[E,F,γ$5$],E]
[error]  match expected type scalaz.Monad[[γ$45$]scalaz.EitherT[E,F,γ$45$]]
[error]     def monadReader[E, R, F[_]](implicit F0: MonadReader[F, R]) = Monad[EitherT[E, F, ?]]
[error]                                                                        ^
```

`eitherTMonadReader` and `eitherTMonadError` ambiguous if add following test in EitherTTest.

```scala
def monadReader[E, R, F[_]](implicit F0: MonadReader[F, R]) = Monad[EitherT[E, F, ?]]
```

Reply:
Reorganised `eitherTMonadReader` instance to resolve the ambiguity error. Please check.